docs/se/guides/40_webclient.adoc

```diff
@@ -0,0 +1,420 @@
+                                       |.Timer;
+                                       |.ConcurrentGauge;
+
+...
+
+public static void main(String[] args) {
+
+    MetricRegistry metricFactory = RegistryFactory.getInstance()
+            .getRegistry(MetricRegistry.Type.APPLICATION);
+
+    String metricName = "metric.GET.localhost";             // <1>
+
+    Counter counter = metricFactory|.counter(metricName);   // <2>
+                                   |.meter(metricName)
+                                   |.timer(metricName)
+                                   |.concurrentGauge(metricName)
+
+    WebClientService clientServiceMetric = WebClientMetrics|.counter()
+                                                           |.meter()
+                                                           |.timer()
+                                                           |.gaugeInProgress()
+
+                    .methods(Http.Method.GET)           // OPTIONAL
+                    .success(true)                      // OPTIONAL
+                    .errors(true)                       // OPTIONAL
+                    .description("Metric Description")  // OPTIONAL
+                    .nameFormat("counter.%1$s.%2$s")
+                    .build();                           // <3>
+----
+<1> Choose the metric name.
+<2> Create a metric from `metricFactory`.
+<3> Build a Web Client Service for counting the GET requests.
+
+The metric name can indicate what is measured. In this example, the metric target GET requests on the localhost.
+In order to pass this information to the webclient, the `nameFormat` method extracts it from the metric name.
+Otherwise, the metric name can also have nothing in common with its job. In this case, the methods with `OPTIONAL`
+comment are not required to be used. The `methods` will target the Http request type of your choice. `success` and `error` will
+respectively measure request if they are successful or failed, and `description` add a metric description.
+
+[source,java]
+.Add the metric service to the WebClient:
+----
+WebClient webClient = WebClient.builder()
+                .baseUri("http://localhost:8080")
+                .addMediaSupport(JsonpSupport.create())
+                .addService(clientServiceMetric)           // <1>
+                .build();
+----
+<1> Register the metric service to the webclient.
+
+Simply use the `addService` method to add the metric to the WebClient on which the metrics will be measured.
+
+[source,java]
+.Print the metric count at the end of the main method:
+----
+System.out.println(metricName + ": " + counter.getCount());
+----
+
+To quickly check metrics are set up correctly, print the counter at the end of the main method. In this guide, the
+WebClient uses GET and PUT requests, so metrics can be applied on.
+
+==== Set up metrics with configuration files
+
+Using configuration file can reduce the code complexity and makes the metrics simpler to use. There is no
```

Comment:
Using the configuration file can reduce the code complexity and make the metrics simpler to use.